lib/src/data/communication/api/network_request.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:allthenews/src/data/communication/api/request.dart';
+import 'package:allthenews/src/di/injector.dart';
+import 'package:allthenews/src/domain/authorization/api_key_repository.dart';
+import 'package:dio/dio.dart';
+import 'package:flutter/foundation.dart';
+import 'package:pretty_dio_logger/pretty_dio_logger.dart';
+
+abstract class _Constants {
+  static const logPrintWidthSize = 300;
+}
+
+class NetworkRequest {
+  final ApiKeyRepository _apiKeyRepository = inject<ApiKeyRepository>();
```

Comment:
hmm... wcześniej napisałeś, że nie będzie tu innej opcji niż wstrzyknięcie do pola, ale przecież apiKeyRepository można przekazywać w konstruktorze razem z baseUrl - bo teraz nawet jak zmienisz baseUrl (a co za tym idzie, pewnie będzie chodzić o skorzystanie z innego api) to przez to wstrzykiwanie w pole nie będzie możliwe skonfigurowanie odpowiedniego apiKeyRepository, bo dla innego baseUrl apiKey też będzie inny :)

Reply:
co do apiKey racja, co do baseUrl to zrobiłem na początku wersję ze wstrzykiwaniem w postaci rejestrowania fabryki w getIt, która przyjmuje parametry, ale de facto efekt końcowy był ten sam - wstrzykiwanie obiektu to było tworzenie nowej instancji klasy z nowym baseUrl, więc stwierdziłem, że równie dobrze można tworzyć ten obiekt w tej klasie. To tyczy się tworzenia obiektu `NetworkRequest` w klasie `NYTimesRestRepository`, to jest powiązane więc zdecydowałem się skomentować tu

Reply:
zawsze można zarejestrować jako singleton, bo i po co konfigurować Dio za każdym razem ;)

Reply:
bo mając przykładowo 2 backendy to raczej nie chcemy już rejestrować `NetworkRequest` jako singleton tylko umożliwić tworzenie nowej instancji tej klasy z innym apiKey i baseUrlem. Co do baseUrla to też repo jest dość dobrym miejscem, żeby je trzymać razem ze wszystkimi innymi ścieżkami do danego backendu niż gdzieś na zewnątrz

Reply:
ale jestem ciekawy co o tym myślisz

Reply:
imo wstrzykiwanie miałoby tu sens, gdybyśmy chcieli sam `NetworkRequest` jeszcze tworzyć inny dla każdego z backendów, jakieś inne konfiguracje tego klienta http i wtedy można by było, tak jak w innych przypadkach, wydzielić interfejs tej klasy i zapodawać różne implementacje. W przypadku gdy mamy taką samą konfigurację dio dla każdego backendu nie bardzo widzę zaletę delegowania tworzenia tych obiektów nad tworzeniem ich w klasie, w której są używane

Reply:
jak będą 2 api to mogą być 2 HttpClienty singletony, każdy odpowiednio skonfigurowany i w zależności które repo jakiego api będzie potrzebowało, tego httpClienta mu się dostarczy :) Nie widzę problemu, żeby url do api był zadeklarowany tam gdzie DI, ale zawsze można zrobić interfejs ApiConfiguration, który by miał pole baseUrl i implementacje by sobie nadpisywały, wtedy HttpClient (NetworkRequest) przyjmowałby 2 parametry, apiConfiguration i apiKeyRepository - wtedy to czysto na maxa :D

Reply:
Z tą konfiguracją Dio się zgadzam, słabe jest kilkakrotne konfigurowanie dio, tylko spójrz, że  teraz przy tworzeniu ręcznie NetworkRequesta w repozytorium jest ten problem na większą skalę, bo jak drugie repozytorium będzie korzystać z tego samego api to w ogóle utworzy się drugi raz identyczny obiekt NetworkRequest ;) Nie powiedziałbym, ze mamy taką samą konfigurację dla każdego backendu, bo i tak w przypadku drugiego backendu dio trzeba skonfigurować drugi raz bo baseUrl i apiKey są tymi zmiennymi, które trzeba skonfigurować, więc jedyną stałą jest DioLogger, ofc, możnaby bawić się w coraz to większe abstrakcje ale imo nie ma sensu i to co zaproponowałem ogranicza ten problem :)

Reply:
zastanawiałem się nad zasadnością tworzenia dwóch singletonów tej samej klasy ale tu w punkt odpowiedź, że jak najbardziej ma to sens, wszystko jasne :)